RNTester/Podfile

```diff
@@ -125,6 +125,13 @@ post_install do |installer|
         puts '  adding arm64e to ' + config.name
       end
     end
+    # TODO(macOS ISS#2323203): the internal Microsoft build pipeline needs macOS arm64 slices
```

Comment:
I am definitely very naive about how this stuff is supposed to work, but if someone takes a dependency on react-native they don't necessarily take a dependency on RNTester, right? So how would a fix in RNTester's podfiles cause clients to end up with arm64 slices? Should we be doing this somewhere in the shared libraries instead?

Reply:
A consumer of the repo will have dependencies on the `*.podspec`'s in this repo.  The `Podfile` in the RNTester folder is only used by the test app here.   External repos would have to *replicate* what the RNTester `Podfile` is doing *if* they wanted macOS arm64 fat slices ahead of released Xcode defaults.

Reply:
So this proves out that RN can build fat binaries. A follow up change has to go in to enable that in CI loops and then any consumers could use either the x86_64 bits or the arm64 bits in their project.

Reply:
But before this week, updating the CI wouldn’t have successful builds for arm64- this is the final evidence that we can in fact run RN on a silicon machine.

Reply:
There was also a fix earlier this week that we needed just to get Xcode 12 universal building for either architecture in the first place